File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string>encode{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        unordered_set<string>unique;
        
        for(int i=0;i<words.size();i++){
            string t="";
            for(int j=0;j<words[i].size();j++){
                int n=words[i][j]-'a';
                t+=encode[n];
            }
            unique.insert(t);
        }
        return unique.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        ArrayList<String>list=new ArrayList<>(Arrays.asList(".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."));
        
        HashSet<String>unique=new HashSet<String>();
        
        for(int i=0;i<words.length;i++){
            String t="";
            for(int j=0;j<words[i].length();j++){
                char c=words[i].charAt(j);
                int n=c-'a';
                t+=list.get(n);
            }
            unique.add(t);
        }
        return unique.size();
    }
}